src/base/QXmppMessage.cpp

```diff
@@ -320,6 +321,32 @@ void QXmppMessage::setParentThread(const QString &parent)
     d->parentThread = parent;
 }
 
+///
+/// Returns the ID of this message's sender's public long-term key.
+///
+/// The key ID can be used by trust management protocols such as
+/// \xep{0450, Automatic Trust Management (ATM)}.
+///
+/// \return the ID of the sender's key
+///
+QString QXmppMessage::senderKey() const
+{
+    return d->senderKey;
+}
+
+///
+/// Sets the ID of this message's sender's public long-term key.
+///
+/// The key ID must be set by the encryption protocol that decrypts this
+/// message such as \xep{0384, OMEMO Encryption}.
+///
+/// \param keyId ID of the sender's key
+///
+void QXmppMessage::setSenderKey(const QString &keyId)
+{
+    d->senderKey = keyId;
+}
+
```

Comment:
Please explain that the key is not parsed/serialized, but is set when an e2ee message is decrypted.